tests/test_premium.py

```diff
@@ -0,0 +1,422 @@
+
+        handle_subscription_updated(subscription)
+
+        assert user.tier_id == BUSINESS_TIER
+
+
+def test_handle_subscription_updated_downgrade(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.id = "sub_123"
+        subscription.status = StripeSubscriptionStatusEnum.CANCELED
+        subscription.cancel_at_period_end = True
+        subscription.current_period_end = datetime.now() + timedelta(days=30)
+        subscription.current_period_start = datetime.now()
+
+        handle_subscription_updated(subscription)
+
+        assert user.tier_id == FREE_TIER
+
+
+def test_handle_subscription_deleted(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = MagicMock()
+        subscription.id = "sub_123"
+
+        handle_subscription_deleted(subscription)
+
+        assert user.tier_id == FREE_TIER
+        assert user.stripe_subscription_id is None
+
+
+def test_handle_invoice_created(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_customer_id = "cus_123"
+        db.session.add(user)
+        db.session.commit()
+
+        handle_invoice_created(
+            MagicMock(
+                id="inv_123",
+                customer="cus_123",
+                hosted_invoice_url="https://example.com",
+                total=2000,
+                status=StripeInvoiceStatusEnum.OPEN,
+                lines=MagicMock(data=[MagicMock(plan=MagicMock(product="prod_123"))]),
+            )
+        )
+
+        stripe_invoice = db.session.query(StripeInvoice).filter_by(invoice_id="inv_123").first()
+        assert stripe_invoice is not None
+
+
+def test_handle_invoice_updated(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_customer_id = "cus_123"
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        invoice = MagicMock(
+            id="inv_123",
+            customer="cus_123",
+            hosted_invoice_url="https://example.com",
+            total=2000,
+            status=StripeInvoiceStatusEnum.OPEN,
+            lines=MagicMock(data=[MagicMock(plan=MagicMock(product="prod_123"))]),
+            subscription="sub_123",
+        )
+        handle_invoice_created(invoice)
+
+        stripe_invoice = db.session.query(StripeInvoice).filter_by(invoice_id="inv_123").first()
```

Comment:
`one()`

Reply:
Fixed in dd6824ec698dfdda90ef67d3f5a60fb331ddcf1a